File: M02/Assignment/SprintChallenge3_PokeApi/app/src/main/java/com/lambdaschool/sprintchallenge3_pokeapi/Pokemon.java
package com.lambdaschool.sprintchallenge3_pokeapi;

import android.graphics.Bitmap;
import android.media.Image;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;

public class Pokemon implements Serializable {
    private int               id;
    private ArrayList<String> moves;
    private String            name, typeA, typeB, spriteUrl;

    public Pokemon(String name) {
        this.id = 0;
        this.moves = new ArrayList<>();
        this.name = name;
        this.typeA = "";
        this.typeB = "";
        this.spriteUrl = "";
    }

    public Pokemon(JSONObject json) {
        try {
            this.id = json.getInt("id");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        try {
            this.name = json.getString("name");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        try {
            this.spriteUrl = json.getJSONObject("sprites").getString("front_default");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        try {
            JSONArray typesArray = json.getJSONArray("types");
            this.typeA = typesArray.getJSONObject(0).getJSONObject("type").getString("name");
            this.typeB = typesArray.getJSONObject(1).getJSONObject("type").getString("name");
        } catch (JSONException e) {
            e.printStackTrace();
        }
        try {
            this.moves = new ArrayList<>();
            JSONArray movesArray = json.getJSONArray("moves");
            for (int i = 0; i < movesArray.length(); ++i) {
                this.moves.add(movesArray.getJSONObject(i).getJSONObject("move").getString("name"));
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public int getId() {
        return id;
    }

    public ArrayList<String> getMoves() {
        return moves;
    }

    public String getName() {
        return name;
    }

    public String getTypeA() {
        return typeA;
    }

    public String getTypeB() {
        return typeB;
    }

    public Bitmap getSprite() {
        return NetworkAdapter.getBitmapFromURL(this.spriteUrl);
    }
}

File: M02/Assignment/SprintChallenge3_PokeApi/app/src/main/java/com/lambdaschool/sprintchallenge3_pokeapi/model/Pokemon.kt
package com.lambdaschool.sprintchallenge3_pokeapi.model

import android.graphics.Bitmap
import com.google.gson.annotations.SerializedName
import com.lambdaschool.sprintchallenge3_pokeapi.NetworkAdapter
import java.io.Serializable

import java.util.ArrayList
// id, moves, name, TypeA, TypeB, sprite get() = NetworkAdapter.getBitmapFromURL(this.spriteUrl!!), spriteUrl
data class Pokemon(
        var id: Int?,
        var moves: String?,
        var name: String?,
        var types: ArrayList<String>?,
        var sprites: ArrayList<String>?
): Serializable {
    val frontDefaultSpriteUrl = sprites?.get(4)

    val typeA: String? = types.let {

        if (it != null) {
            if (it.size >= 1) {
                return@let it[0]
            } else {
                return@let null
            }
        } else return@let null
    }

    val typeB: String? = types.let {
        if (it != null) {
            if (it.size >= 2) {
                return@let it[1]
            } else {
                return@let null
            }
        } else return@let null
    }

    fun spriteBitmap() : Bitmap? {
        return sprites?.get(0)?.let { NetworkAdapter.getBitmapFromURL(it) }
    }

}
